tests/simplify.py

```diff
@@ -159,20 +157,44 @@ def record_node(self, input_id, is_sample=False):
         self.node_id_map[input_id] = output_id
         return output_id
 
+    def unrecord_node(self, input_id, output_id):
```

Comment:
`rewind_node` might be a better name because this only works on the last node recorded.